distributed/shuffle/_rechunk.py

```diff
@@ -252,3 +264,226 @@ def split_axes(old: ChunkedAxes, new: ChunkedAxes) -> SplitAxes:
             old_chunk.sort(key=lambda split: split.slice.start)
         axes.append(old_axis)
     return axes
+
+
+def convert_chunk(data: bytes) -> np.ndarray:
```

Comment:
How do these type annotations work for normal execution. AFAICT, `np` is not imported at the top-level, so at runtime (rather than type checking time) this name is not defined?

Reply:
They are not interpreted at runtime thanks to `from __future__ import annotations`.

> Note: If `from __future__ import annotations` is used, annotations are not evaluated at function definition time. Instead, they are stored as strings in `__annotations__`. This makes it unnecessary to use quotes around the annotation (see [PEP 563](https://peps.python.org/pep-0563/)).

https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING